src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+            int executeUpdate = preparedStatement.executeUpdate();
+            if (executeUpdate < 1) {
+                throw new DataProcessingException("Can not process");
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant create new book ", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(parseBook(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve list of books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            int executeUpdate = preparedStatement.executeUpdate();
+            if (executeUpdate > 0) {
+                return book;
+            }
+            throw new DataProcessingException("No book found with id " + book.getId());
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to update book: " + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int executeUpdate = preparedStatement.executeUpdate();
+            return executeUpdate > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to delete book by id " + id, e);
```

Comment:
The error message could be more descriptive. Consider including more context, such as 'Failed to execute deleteById query for book with id ' + id'.